server/maintenance_requests.go

```diff
@@ -0,0 +1,56 @@
+// This file provides NewApp() with functions for maintenance requests
+package main
+
+import(
+	"github.com/gin-gonic/gin"
+	"github.com/a2fumn2022/gpmn-nsl/server/datastore"
+	"time"
+	"io/ioutil"
+	"net/http"
+	"encoding/json"
+)
+
+/*type CreateRequest struct {
+	Request string `json:"request"`
+	UserSubmitted int `json:"id"`
+	DateSubmitted time.Time `json:"date"`
+}*/
+
+/* AllMaintenanceRequests returns all the maintenance_requests from the database
+*  AllMaintenanceRequests is a function of app */
+func (app *App) AllMaintenanceRequests(c *gin.Context) {
+	/* Get data using psql.AllMaintenanceRequests and pass in app.db for the
+	* connection */
+
+	/* Takes the data returned in the psql func and formats it to a valid format
+	* using the gin.H func. See if it you can just use a Struct */
+
+	/* Use the gin.context.JSON func to return a status and the data. See auth.go
+	* for an example */
+}
+
+func (app *App) CreateMaintenanceRequests(c *gin.Context) {
+	byteData, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	var newRequest datastore.MaintenanceRequest
+	err = json.Unmarshal(byteData, &newRequest)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	if newRequest.Request == "" {
+		c.JSON(http.StatusNotFound, gin.H{"error": "Request field is empty or missing and is needed"})
```

Comment:
Last change! We need to provide the right error. Right not you're sending back a `404` message, which means "Not found." But what you really want to give back is a `400` Bad Request error.

Here's a useful cheat sheet: https://devhints.io/http-status
The cheatsheet explanation of `400`: https://httpstatuses.com/400
Here's the Golang `net/http` list of errors: https://golang.org/src/net/http/status.go

And for fun: https://httpstatuses.com/418

Reply:
Let's also change the text to read:

"Check that the `request` field is not blank or missing."

Reply:
fix: used error 400 instead of 404.
Lol error 418